fact_extractor/plugins/unpacking/draytek_vigor_167/code/draytek_vigor_167.py

```diff
@@ -0,0 +1,35 @@
+'''
+This plugin extracts the squashfs root filesystem in Draytek Vigor 167 firmware containers.
+'''
+import struct
+from pathlib import Path
+from common_helper_files import write_binary_to_file
+
+NAME = 'Draytek Vigor 167'
+MIME_PATTERNS = ['firmware/draytek-vigor-167']
+VERSION = '0.1'
+
+header_layout = '> 4s I I I 64x I I 49x I I 117x'
+header = struct.Struct(header_layout)
+
+
+def unpack_function(file_path, tmp_dir):
+    '''
+    file_path specifies the input file.
+    tmp_dir must be used to store the extracted files.
+    Optional: Return a dict with meta information
+    '''
+    with open(file_path, 'rb') as f:
+        signature = header.unpack(f.read(header.size))
+        f.seek(signature[1] + signature[4])
+        squashfs = f.read(signature[5])
+
+    output_file_path = Path(tmp_dir) / 'squashfs_root'
+    write_binary_to_file(squashfs, output_file_path)
+    return {'output': 'successfully unpacked image'}
```

Comment:
We could also fully unpack the signature and append it to the returned dict here as supplementary meta data.

Reply:
Good idea - can i just append the returned dict with a `signature` key or should i return the signature inside the `output` key?

Reply:
Appended to return dict. Changed in 76e86b2f0231d96c9ad69edb76677900c56eeec5